cadc-pkg-server/src/main/java/org/opencadc/pkg/server/TarWriter.java

```diff
@@ -69,88 +69,34 @@
 
 package org.opencadc.pkg.server;
 
-import ca.nrc.cadc.io.MultiBufferIO;
-import ca.nrc.cadc.net.HttpGet;
-import ca.nrc.cadc.net.ResourceAlreadyExistsException;
-import ca.nrc.cadc.net.ResourceNotFoundException;
-import ca.nrc.cadc.net.TransientException;
-import java.io.IOException;
-import java.io.InputStream;
 import java.io.OutputStream;
-import java.net.URL;
 import java.util.Date;
 import org.apache.commons.compress.archivers.ArchiveEntry;
 import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
 import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
 import org.apache.log4j.Logger;
 
-public class TarWriter {
+public class TarWriter extends PackageWriter {
     private static final Logger log = Logger.getLogger(TarWriter.class);
 
-    private final TarArchiveOutputStream tout;
-    
-    public TarWriter(OutputStream ostream) {
-        this.tout = new TarArchiveOutputStream(ostream);
-        tout.setLongFileMode(TarArchiveOutputStream.LONGFILE_POSIX);
-    }
+    public static final String MIME_TYPE = "application/x-tar";
+    public static final String EXTENSION = ".tar";
 
-    public void close() throws IOException {
-        tout.finish();
-        tout.close();
+    public TarWriter(OutputStream ostream) {
+        super(new TarArchiveOutputStream(ostream));
+        ((TarArchiveOutputStream)super.aout).setLongFileMode(TarArchiveOutputStream.LONGFILE_POSIX);
```

Comment:
From the discussion, yeah this is a little yuck, but for a one liner it's OK. My main concern with it is that it requires `aout` to be protected rather than private in the base class.

The alternative would be to have a no-arg PackageWriter() and require (document) subclasses to assign to `super.aout` before returning from ctor. That doesn't follow the normal practice/style of Writer and OutputStream...

An alternative to documenting "assign to aout" would be to have an  `abstract ArchiveOutputStream getArchiveOutputStream()` in PackageWriter() that was called in no-arg PackageWriter()... *and* make `aout` private. My IDE complains about ctors calling overridable methods, so someone thinks that a poor programming style... but it does get you out of the issue of having to call super(...) first ickiness and allows `aout` to be private without losing clarity on what subclasses have to do.

this is all optional: the current one-extra line is OK for now

Reply:
Ok. Leaving it as is. Even changing this to getAout() wouldn't escape the ugly casing going on there.